[BUG] Integration cache_test failures - ArrayIndexOutOfBoundsException

**Describe the bug**
The integration cache_test is failing:

[2021-11-02T14:30:15.572Z] FAILED ../../src/main/python/cache_test.py::test_cache_join[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'true'}-Left-String][IGNORE_ORDER]
.... lots of others as well

```
2021-11-02T14:01:19.239Z] 21/11/02 14:01:19 WARN TaskSetManager: Lost task 4.0 in stage 211.0 (TID 1138) (10.233.92.210 executor 0): java.lang.ArrayIndexOutOfBoundsException: 0
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.HostToGpuCoalesceIterator.$anonfun$addBatchToConcat$1(HostColumnarToGpu.scala:330)
[2021-11-02T14:01:19.239Z]  at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.HostToGpuCoalesceIterator.addBatchToConcat(HostColumnarToGpu.scala:329)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.AbstractGpuCoalesceIterator.addBatch(GpuCoalesceBatches.scala:408)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.AbstractGpuCoalesceIterator.$anonfun$next$1(GpuCoalesceBatches.scala:336)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.AbstractGpuCoalesceIterator.withResource(GpuCoalesceBatches.scala:202)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.AbstractGpuCoalesceIterator.next(GpuCoalesceBatches.scala:322)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.AbstractGpuCoalesceIterator.next(GpuCoalesceBatches.scala:202)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.GpuHashAggregateIterator.aggregateInputBatches(aggregate.scala:282)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.GpuHashAggregateIterator.$anonfun$next$2(aggregate.scala:237)
[2021-11-02T14:01:19.239Z]  at scala.Option.getOrElse(Option.scala:189)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:234)
[2021-11-02T14:01:19.239Z]  at com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:180)
[2021-11-02T14:01:19.239Z]  at org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBase$$anon$1.partNextBatch(GpuShuffleExchangeExecBase.scala:291)
[2021-11-02T14:01:19.239Z]  at org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBase$$anon$1.hasNext(GpuShuffleExchangeExecBase.scala:307)
[2021-11-02T14:01:19.239Z]  at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:140)
[2021-11-02T14:01:19.239Z]  at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
[2021-11-02T14:01:19.239Z]  at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
[2021-11-02T14:01:19.239Z]  at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
[2021-11-02T14:01:19.239Z]  at org.apache.spark.scheduler.Task.run(Task.scala:131)
[2021-11-02T14:01:19.239Z]  at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:506)
[2021-11-02T14:01:19.239Z]  at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1462)
[2021-11-02T14:01:19.239Z]  at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:509)
[2021-11-02T14:01:19.239Z]  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2021-11-02T14:01:19.239Z]  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
```

--- a/sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/v2/Spark32XShims.scala
+++ b/sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/v2/Spark32XShims.scala
@@ -642,8 +642,10 @@ trait Spark32XShims extends SparkShims {
             + TypeSig.ARRAY).nested(), TypeSig.all),
         (scan, conf, p, r) => new SparkPlanMeta[InMemoryTableScanExec](scan, conf, p, r) {
           override def tagPlanForGpu(): Unit = {
-            if (!scan.relation.cacheBuilder.serializer.isInstanceOf[ParquetCachedBatchSerializer]) {
-              willNotWorkOnGpu("ParquetCachedBatchSerializer is not being used")
+            scan.relation.cacheBuilder.serializer match {
+              case _: com.nvidia.spark.ParquetCachedBatchSerializer => ()
+              case _ =>
+                willNotWorkOnGpu("ParquetCachedBatchSerializer is not being used")
             }
           }
 
@@ -885,7 +887,7 @@ trait Spark32XShims extends SparkShims {
       exportColumnRdd: Boolean): GpuColumnarToRowExecParent = {
     val serName = plan.conf.getConf(StaticSQLConf.SPARK_CACHE_SERIALIZER)
     val serClass = ShimLoader.loadClass(serName)
-    if (serClass == classOf[ParquetCachedBatchSerializer]) {
+    if (serClass == classOf[com.nvidia.spark.ParquetCachedBatchSerializer]) {
       GpuColumnarToRowTransitionExec(plan)
     } else {
       GpuColumnarToRowExec(plan)

